evap/staff/importers.py

```diff
@@ -225,31 +226,38 @@ def __init__(self):
 
     def read_book(self, file_content):
         try:
-            self.book = xlrd.open_workbook(file_contents=file_content)
-        except xlrd.XLRDError as e:
+            self.book = openpyxl.load_workbook(BytesIO(file_content))
+
+        except Exception as e:  # pylint: disable=broad-except
             self.errors[ImporterError.SCHEMA].append(_("Couldn't read the file. Error: {}").format(e))
 
     def check_column_count(self, expected_column_count):
-        for sheet in self.book.sheets():
-            if sheet.nrows <= self.skip_first_n_rows:
+        for sheet in self.book:
+            if sheet.max_row <= self.skip_first_n_rows:
                 continue
-            if sheet.ncols != expected_column_count:
+            if sheet.max_column != expected_column_count:
                 self.errors[ImporterError.SCHEMA].append(
                     _("Wrong number of columns in sheet '{}'. Expected: {}, actual: {}").format(
-                        sheet.name, expected_column_count, sheet.ncols
+                        sheet.title, expected_column_count, sheet.max_column
                     )
                 )
 
     def for_each_row_in_excel_file_do(self, row_function):
-        for sheet in self.book.sheets():
+        for sheet in self.book:
             try:
-                for row in range(self.skip_first_n_rows, sheet.nrows):
+                for rowIdx, row in enumerate(
+                    sheet.iter_rows(min_row=self.skip_first_n_rows + 1, values_only=True), self.skip_first_n_rows
+                ):
                     # see https://stackoverflow.com/questions/2077897/substitute-multiple-whitespace-with-single-whitespace-in-python
-                    row_function([" ".join(cell.split()) for cell in sheet.row_values(row)], sheet, row)
-                self.success_messages.append(_("Successfully read sheet '%s'.") % sheet.name)
+                    row_function(
+                        [" ".join(cell.split()) if cell is not None else "" for cell in row],
+                        sheet,
+                        rowIdx,
+                    )
```

Comment:
The `enumerate`, `skip`, `min_row` thing feels a bit duplicated, how do you feel about this version:
```python
for rowIdx in range(self.skip_first_n_rows, sheet.max_row):
    values = [cell.value or "" for cell in sheet[rowIdx + 1]]
    # see https://stackoverflow.com/questions/2077898/substitute-multiple-whitespace-with-single-whitespace-in-python
    cleaned_values = [" ".join(value.split()) for value in values]
    row_function(cleaned_values, sheet, rowIdx)
```